helm-preflights/preflights.sh

```diff
@@ -0,0 +1,296 @@
+#!/bin/bash
+
+script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+function echo_pass() {
+  echo -e "\033[1;32mPASS\033[0m"
+}
+
+function echo_ko() {
+  echo -e "\033[1;31mKO\033[0m "
+}
+
+function echo_error() {
+  echo -en "\033[1;31m[ERROR]\033[0m " >&2
+  echo $@ >&2
+}
+
+function echo_warn() {
+  echo -en "\033[1;32m[WARN]\033[0m "
+  echo $@
+}
+
+function install_preflight() {
+  echo -e "--- INSTALLING PREFLIGHT PLUGIN"
+  curl --silent https://krew.sh/preflight | bash >$script_dir/preflights_install.logs 2>&1
+}
+
+function usage() {
+  cat <<USAGE
+
+Usage:
+    $(basename $0) [OPTIONS] chart
+
+Description:
+    Execute Gitguardian Preflight on a cluster.
+    Chart can be local path or OCI uri.
+    Dependencies: kubectl, helm, preflight plugin (installable using --install-preflight option) 
+
+OPTIONS:
+
+    -f FILE
+        Pass a values file, can be used several times
+
+    -n NAMESPACE
+        Specify the Kubernetes destination namespace
+        
+    --local
+        Execute only local tests
+
+    --install-preflight
+        Install latest preflight plugin using krew for local preflights
+
+    --remote
+        Execute only remote tests
+
+    --force
+        Force retemplating even if preflights have been played before     
+
+    --nosave
+        Do not save results in-cluster (not recommended)            
+
+    -h | --help
+        Display this help message
+USAGE
+}
+
+#conf
+REMOTE_PREFLIGHTS_TEMPLATE="templates/on-prem/helm_preflights_remote.yaml"
+LOCAL_PREFLIGHTS_TEMPLATE="templates/on-prem/helm_preflights_local.yaml"
+
+#inputs
+CHART=""
+NAMESPACE=""
+LOCAL_CHECKS="yes"
+REMOTE_CHECKS="yes"
+VALUES_FILES=""
+FORCE="no"
+SAVE="yes"
+INSTALL_PREFLIGHT="no"
+
+#outputs
+GLOBAL_OUTPUT=""
+GLOBAL_RC=0
+LOCAL_CHECKS_STATUS="empty"
+REMOTE_CHECKS_STATUS="empty"
+
+while (("$#")); do
+  case "$1" in
+  --local)
+    REMOTE_CHECKS="no"
+    shift
+    ;;
+  --remote)
+    LOCAL_CHECKS="no"
+    shift
+    ;;
+  -n)
+    shift
+    NAMESPACE="--namespace $1"
+    shift
+    ;;
+  --force)
+    FORCE="yes"
+    shift
+    ;;
+  --install-preflight)
+    INSTALL_PREFLIGHT="yes"
+    shift
+    ;;
+  --nosave)
+    SAVE="no"
+    shift
+    ;;
+  -f)
+    shift
+    VALUES_FILES+="-f $1 "
+    shift
+    ;;
+  -h | --help)
+    usage
+    exit 0
+    ;;
+  *) # positional argument
+    CHART=$1
```

Comment:
You could add a test here to check that `CHART` var is not already set, and if this is the case raise an error:

At the moment, the script allows to specify multiple `args` and the last one will be used to define CHART....